components/concept/storage/src/main/java/mozilla/components/concept/storage/CreditCardsAddressesStorage.kt

```diff
@@ -315,7 +315,7 @@ data class UpdatableCreditCardFields(
 @SuppressLint("ParcelCreator")
 @Parcelize
 data class Address(
-    val guid: String,
+    val guid: String? = null,
```

Comment:
I am not sure about this change. The `Address` here is strictly meant to be 1:1 to the AS data structure defined in https://github.com/mozilla/application-services/blob/main/components/autofill/src/autofill.udl#L62 and seeing the need to use `!!` while working with the AS storage compounds my fear that this is not what we want.

We actually have a `CreditCardEntry` and `LoginEntry` now which just represents that data object that is passed between the AC and GV. So, I would prefer if we just added a `AddressEntry` which will be used for the autocompletion between AC and GV.

Reply:
Circling back to this, I am wondering if we could just provide a default guid value to when going from GV <-> AC. That way we can avoid making an `AddressEntry`

Reply:
I am gonna sleep on this. It's possible I am overthinking this nullability of guid.

Reply:
Ok thought this over, there should be no consequences when converting the Autocomplete.Address to Address if we default the guid to ""